### DISCUSSION TITLE
[Bug]: Cookies are missing in `request` for loaders and actions in CSR

### DISCUSSION THREAD

**Discussion by rmedaer** [remix-run/react-router#10825] [Proposals]
### What version of React Router are you using?

6.15.0

### Steps to Reproduce

Create a _Route_ with a _loader_ (in CSR) and get `Cookie` header from the request.

```jsx
function loader({ request }) {
  const cookie = request.headers.get('cookie');
  console.log(cookie);

  // (...) business logic using cookies
}

const router = createBrowserRouter({
    path: '/',
    element: <HelloWorld/>,
    loader: loader,
});

// (...)
```

### Expected Behavior

I expect cookies to be available through the `Cookie` header (as they ~are~ might be in SSR) so I can use the same piece of code for both CSR and SSR.

### Actual Behavior

The cookies are not available in the `Cookie` header.

Analyzing the `createClientSideRequest` function in [`react-router/packages/router/router.ts`](https://github.com/remix-run/react-router/blob/496b1fe8253643171ecca6e6a945d98386c4eb00/packages/router/router.ts#L3809C1-L3809C34), it looks like the cookies (from [`document.cookie`](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie)) are not included in the `Request` created:

```ts
function createClientSideRequest(
  history: History,
  location: string | Location,
  signal: AbortSignal,
  submission?: Submission
): Request {
  let url = history.createURL(stripHashFromPath(location)).toString();
  let init: RequestInit = { signal };

  // (...) 

  return new Request(url, init);
}
```

### Notes

It might be a feature request but since it's available in SSR, I expect the same behavior in CSR.

**Comment by brophdawg11** [comment]
I'm going to move this to a feature request discussion.

IMO, Cookies are inherently different in CSR and SSR due to cookie paths, origins, http only, etc., so I think it's probably better/safer to read from the `Request` in SSR and `document.cookie` in CSR.  It feels like it would be error-prone to try to fill in the request cookies from `document.cookie` and share the same code, but we can open this as a proposal and see what other's think.

**Reply by VISHALl1166** [reply]
i can receive header cookies in req.headers. please tell me how can i send these cookies with loader function fetch request
